### PR TITLE
Case insensitive routing

### PR DESCRIPTION
**Pull request by Joannis**
#208 related

### CODE REVIEW

File: Sources/Routing/TrieRouter.swift [hunk 1]
```diff
@@ -53,8 +57,12 @@ public final class TrieRouter<Output> {
                 case .constant(let data, let size):
                     // if we find a constant route path that matches this component,
                     // then we should use it.
-                    let match = component.withByteBuffer { buffer in
-                        return memcmp(data, buffer.baseAddress!, size) == 0
+                    let match = component.withByteBuffer { buffer -> Bool in
+                        if self.caseInsensitive {
+                            return data.caseInsensitiveEquals(to: buffer)
+                        } else {
+                            return memcmp(data, buffer.baseAddress!, size) == 0
```
Comment (tanner0101) [inline Sources/Routing/TrieRouter.swift]: @gwynne is there a more ABI-safe way to do this?

Comment (gwynne) [reply Sources/Routing/TrieRouter.swift]:
```swift
return data.elementsEqual(buffer)
```

Comment (kevinup7) [reply Sources/Routing/TrieRouter.swift]: In addition to any safety issues, the use of `memcmp`currently causes a bug where routes with `A` and `AA` will always end up resolving to the `A` route.

Comment (gwynne) [reply Sources/Routing/TrieRouter.swift]: That's less `memcmp()`'s fault per se and more of an ordering logic error.

For the record, `memcmp()` is not really _unsafe_ in this particular usage so much as it is bad style; one should always use the Swift stdlib whenever possible, even when working with raw pointers.

Comment (tanner0101) [reply Sources/Routing/TrieRouter.swift]: @gwynne you're the best! ty

Comment (tanner0101) [reply Sources/Routing/TrieRouter.swift]: @kevinup7 I will add a test case for that. That's a bug

**Pull request merged**